src/main/java/io/leonis/algieba/spatial/BoundaryPotentialField.java

```diff
@@ -0,0 +1,148 @@
+    return this.getPotential(positionVector.getDouble(0, 0))
+        + this.getPotential(this.getWidth() - positionVector.getDouble(0, 0))
+        + this.getPotential(positionVector.getDouble(1, 0))
+        + this.getPotential(this.getLength() - positionVector.getDouble(1, 0));
+  }
+
+  /**
+   * See <a href="http://www.wolframalpha.com/input/?i=1%2F(1+%2B+e%5E(D+-+d))">this equation.</a>
+   *
+   * @param distanceToBoundary The distance to the boundary in mm.
+   * @return The potential due to a single boundary at the specified distance from that boundary.
+   */
+  public double getPotential(final double distanceToBoundary) {
+    return 1 / (1 + Math.exp(this.getFieldDisplacement() - distanceToBoundary));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public INDArray getForce(final INDArray positionVector) {
+    return Vectors.columnVector(
+        -1 * this.getForceMagnitude(positionVector.getDouble(0, 0)),
+        0)
+        .add(Vectors.columnVector(
+            this.getForceMagnitude(this.getWidth() - positionVector.getDouble(0, 0)),
+            0))
+        .add(Vectors.columnVector(
+            0,
+            -1 * this.getForceMagnitude(positionVector.getDouble(1, 0))))
+        .add(Vectors.columnVector(
+            0,
+            this.getForceMagnitude(this.getLength() - positionVector.getDouble(1, 0))));
+  }
+
+  /**
+   * See <a href="http://www.wolframalpha.com/input/?i=derivative+of+1%2F(1+%2B+e%5E(D+-+d))">this
+   * equation</a>.
+   *
+   * @param distanceToBoundary The distance from the boundary for which the compute the magnitude of
+   *                           the force vector.
+   * @return The magnitude of the force vector due to a single boundary.
+   */
+  private double getForceMagnitude(final double distanceToBoundary) {
+    return Math.exp(distanceToBoundary + this.getFieldDisplacement())
+        / Math.pow(Math.exp(distanceToBoundary) + Math.exp(this.getFieldDisplacement()), 2);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public double getLineIntegral(final INDArray lowerBound, final INDArray upperBound) {
+    return Math.sqrt(Transforms.pow(upperBound.sub(lowerBound), 2).sumNumber().doubleValue())
+        * (this.computeLineIntegral(lowerBound, upperBound, CardinalDirection.NORTH).apply(1d)
+        + this.computeLineIntegral(lowerBound, upperBound, CardinalDirection.SOUTH).apply(1d)
+        + this.computeLineIntegral(lowerBound, upperBound, CardinalDirection.EAST).apply(1d)
+        + this.computeLineIntegral(lowerBound, upperBound, CardinalDirection.WEST).apply(1d)
+        - this.computeLineIntegral(lowerBound, upperBound, CardinalDirection.NORTH).apply(0d)
+        - this.computeLineIntegral(lowerBound, upperBound, CardinalDirection.SOUTH).apply(0d)
+        - this.computeLineIntegral(lowerBound, upperBound, CardinalDirection.EAST).apply(0d)
+        - this.computeLineIntegral(lowerBound, upperBound, CardinalDirection.WEST).apply(0d));
```

Comment:
what about the intercardinal directions such as  `NORTH_EAST` ?

Reply:
I might also prefer `computeLineIntergralNorth()` for all the methods. that way the switch case and the extra argument aren't necessary.

Reply:
those intercardinal directions would turn this rectangle into an octagon. might need to rename this to `RectangularBoundaryField` then